### ISSUE TITLE
False positive: type="importmap" incorrectly flagged with "must only contain string values" error

### ISSUE THREAD

**Issue by plufz** [validator/validator#1889]
```
 <!DOCTYPE html>
  <html lang="en">
  <head>
      <title>Test Import Map</title>
      <script type="importmap">
      {
          "imports": {
              "app": "/path/to/app.js",
              "lib": "/path/to/lib.js"
          }
      }
      </script>
  </head>
  <body>
  </body>
  </html>
```

> A specifier map defined in a "imports" property within the content of a "script" element with a "type" attribute whose value is "importmap" must only contain string values.

I have tested the exact same html many times before and it has validated. There seems to have been a recent update to https://validator.w3.org/nu/ which changed this.

**Comment by sideshowbarker** [comment]
@tenzap It is possible this is a regression due to the JSON library change?

The relevant source is here: https://github.com/validator/validator/blob/c81924e77329129ec36ab1dee6a728d7336eac38/src/nu/validator/checker/schematronequiv/Assertions.java#L1810-L1817

So apparently, that `instanceof String` check is now failing — despite the values in the import map both being strings.

**Comment by sideshowbarker** [comment]
@plufz Thanks for catching this and taking time report it. I have a patch with a proposed fix at https://github.com/validator/validator/pull/1890. As soon as as we get a fix landed, I’ll push an update to https://validator.w3.org/nu/ and then post a comment here to let you know

**Comment by plufz** [comment]
> [@plufz](https://github.com/plufz) Thanks for catching this and taking time report it. I have a patch with a proposed fix at [#1890](https://github.com/validator/validator/pull/1890). As soon as as we get a fix landed, I’ll push an update to [validator.w3.org/nu](https://validator.w3.org/nu/) and then post a comment here to let you know

Thanks for putting together a patch so quickly!
Really appreciate how fast you came up with a fix!

**Comment by sideshowbarker** [comment]
OK, for the record here, I’ve deployed an update to https://validator.w3.org/nu/, so no errors are reported there any longer for the document in the issue description:

https://validator.w3.org/nu/?showsource=yes&doc=data%3Atext%2Fhtml%3Bcharset%3Dutf-8%2C%253C%2521DOCTYPE%2520html%253E%250A%2520%253Chtml%2520lang%253D%2522en%2522%253E%250A%2520%253Chead%253E%250A%2520%2520%2520%2520%2520%253Ctitle%253ETest%2520Import%2520Map%253C%2Ftitle%253E%250A%2520%2520%2520%2520%2520%253Cscript%2520type%253D%2522importmap%2522%253E%250A%2520%2520%2520%2520%2520%257B%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2522imports%2522%253A%2520%257B%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2522app%2522%253A%2520%2522%2Fpath%2Fto%2Fapp.js%2522%252C%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2520%2522lib%2522%253A%2520%2522%2Fpath%2Fto%2Flib.js%2522%250A%2520%2520%2520%2520%2520%2520%2520%2520%2520%257D%250A%2520%2520%2520%2520%2520%257D%250A%2520%2520%2520%2520%2520%253C%2Fscript%253E%250A%2520%253C%2Fhead%253E%250A%2520%253Cbody%253E%250A%2520%253C%2Fbody%253E%250A%2520%253C%2Fhtml%253E

@plufz Thanks again to you for having taken time to report this

**Comment by plufz** [comment]
🙏 Thank you for deploying this so swiftly.
